core/src/main/java/ai/timefold/solver/core/api/domain/metamodel/PlanningVariableMetaModel.java

```diff
@@ -0,0 +1,51 @@
+package ai.timefold.solver.core.api.domain.metamodel;
+
+import ai.timefold.solver.core.api.domain.variable.PlanningVariable;
+
+/**
+ * A {@link VariableMetaModel} that represents a @{@link PlanningVariable basic planning variable}.
+ *
+ * <p>
+ * <strong>This package and all of its contents are part of the Move Streams API,
+ * which is under development and is only offered as a preview feature.</strong>
+ * There are no guarantees for backward compatibility;
+ * any class, method or field may change or be removed without prior notice,
+ * although we will strive to avoid this as much as possible.
+ * <p>
+ * We encourage you to try the API and give us feedback on your experience with it,
+ * before we finalize the API.
+ * Please direct your feedback to
+ * <a href="https://github.com/TimefoldAI/timefold-solver/discussions">Timefold Solver Github</a>.
+ *
+ * @param <Solution_> the solution type
+ * @param <Entity_> the entity type
+ * @param <Value_> the value type
+ */
+public non-sealed interface PlanningVariableMetaModel<Solution_, Entity_, Value_>
+        extends VariableMetaModel<Solution_, Entity_, Value_> {
+
+    @Override
+    default boolean isList() {
+        return false;
+    }
+
+    @Override
+    default boolean isGenuine() {
+        return true;
+    }
+
+    /**
+     * Returns whether the planning variable allows null values.
+     *
+     * @return {@code true} if the planning variable allows null values, {@code false} otherwise.
+     */
+    boolean allowsUnassigned();
```

Comment:
I wonder if `allowsUnassigned` should be part of `VariableMetaModel`. If we need to cast variable models to check this, perhaps we could avoid that by adding it to the parent class.

Reply:
This is intentional, and is related to the pattern matching that I mentioned in another comment. List variable cannot be chained, and therefore shouldn't confuse people with that option.

That's why we have inheritance - more specific objects carry more specific methods.